org.mwc.cmap.grideditor/src/org/mwc/cmap/grideditor/GridEditorView.java

```diff
@@ -66,13 +76,55 @@ public void createPartControl(final Composite parent)
 		myActions.fillActionBars(actionBars);
 		actionBars.setGlobalActionHandler(ActionFactory.UNDO.getId(), myUndoAction);
 		actionBars.setGlobalActionHandler(ActionFactory.REDO.getId(), myRedoAction);
+		_myPartMonitor =
+        new PartMonitor(getSite().getWorkbenchWindow().getPartService());
+		_myPartMonitor.addPartListener(CorePlotEditor.class, PartMonitor.CLOSED, new PartMonitor.ICallback()
+    {
+      
+      @Override
+      public void eventTriggered(String type, Object instance,
+          IWorkbenchPart parentPart)
+      {
+        if(instance.equals(displayedPlot)) {
+          //set input null now.
+          activePlots.remove(instance);

```

Comment:
@ayeshare - we never use the contents of activePlots.  I think we don't need them. Could you check?